Zend/zend_vm_def.h

```diff
@@ -3404,6 +3404,15 @@ ZEND_VM_HOT_OBJ_HANDLER(112, ZEND_INIT_METHOD_CALL, CONST|TMPVAR|UNUSED|THIS|CV,
 		} while (0);
 	}
 
+	if (OP2_TYPE != IS_CONST) {
+		if (UNEXPECTED(zend_string_equals_literal_ci(Z_STR_P(function_name), ZEND_CONSTRUCTOR_FUNC_NAME))) {
```

Comment:
You'll want to check the CTOR flag instead.

Reply:
Modified to use the flag